mlrun/api/crud/model_endpoints.py

```diff
@@ -977,43 +667,507 @@ def deploy_model_monitoring_batch_processing(
+        start: str = "now-1h",
+        end: str = "now",
+        feature_analysis: bool = False,
+        endpoint_id: str = None,
+        raise_for_status=None,
+    ) -> mlrun.api.schemas.ModelEndpoint:
+        """
+        Get a single model endpoint object. You can apply different time series metrics that will be added to the
+           result.
+
+        :param endpoint_id:      The unique id of the model endpoint.
+        :param start:            The start time of the metrics.
+        :param end:              The end time of the metrics.
+        :param metrics:          A list of metrics to return for the model endpoint. There are pre-defined metrics for
+                                 model endpoints such as predictions_per_second and latency_avg_5m but also custom
+                                 metrics defined by the user. Please note that these metrics are stored in the time
+                                 series DB and the results will be appeared under model_endpoint.spec.metrics.
+        :param feature_analysis: When True, the base feature statistics and current feature statistics will be added to
+                                 the output of the resulting object.
+        :param raise_for_status: Raise a specific error based on the given response status code.
+
+        :return: A ModelEndpoint object.
+        """
+        pass
+
+    def list_model_endpoints(
+        self, model: str, function: str, labels: typing.List, top_level: bool
+    ):
+        """
+        Returns a list of endpoint unique ids, supports filtering by model, function,
+        labels or top level. By default, when no filters are applied, all available endpoint ids for the given project
+        will be listed.
+
+        :param model:           The name of the model to filter by.
+        :param function:        The name of the function to filter by.
+        :param labels:          A list of labels to filter by. Label filters work by either filtering a specific value
+                                of a label (i.e. list("key==value")) or by looking for the existence of a given
+                                key (i.e. "key").
+        :param top_level:       If True will return only routers and endpoint that are NOT children of any router.
+
+        :return: List of model endpoints unique ids.
+        """
+        pass
+
+
+class _ModelEndpointKVStore(_ModelEndpointStore):
+    """
+    Handles the DB operations when the DB target is from type KV. For the KV operations, we use an instance of V3IO
+    client and usually the KV table can be found under v3io:///users/pipelines/project-name/model-endpoints/endpoints/.
+    """
+
+    def __init__(self, access_key: str, project: str):
+        super().__init__(access_key=access_key, project=project)
+        # Initialize a V3IO client instance
+        self.client = mlrun.utils.v3io_clients.get_v3io_client(
+            endpoint=mlrun.mlconf.v3io_api
```

Comment:
Since this gets an `access_key`, it would be best if it is passed to `get_v3io_client` as well instead of passing it in each and every call to the client.

Reply:
Fixed
```
        super().__init__(access_key=access_key, project=project)
        # Initialize a V3IO client instance
        self.client = mlrun.utils.v3io_clients.get_v3io_client(
            endpoint=mlrun.mlconf.v3io_api, access_key=access_key
        )
```